### PR TITLE
[WIP] Add reactjs components

### PR DESCRIPTION
**Pull request by Br3nda**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -167,12 +164,19 @@ group :development, :test do
...
 
 group :test do
+  gem 'capybara'                # integration tests
+  gem 'capybara-email'          # integration tests for email
+  gem 'capybara-screenshot'     # for test debugging
   gem 'codeclimate-test-reporter', require: false
+  gem 'database_cleaner'
   gem 'percy-capybara', '~> 4.0.0'
   gem 'rails-controller-testing'
+  gem 'rspec-activemodel-mocks'
```
Comment (CloCkWeRX) [inline Gemfile]: Defined already on line 152?
> Gem `rspec-activemodel-mocks` requirements already given on line 152 of the Gemfile. (Bundler/DuplicatedGem)

File: app/resources/api/v1/planting_resource.rb [hunk 2]
```diff
@@ -9,42 +9,65 @@ class PlantingResource < BaseResource
...
+      filter :crop_id
+      filter :finished
+      filter :garden
+      filter :garden_id
+      filter :owner
+      filter :owner_id
+      filter :planted_from
+      filter :slug
+      filter :active, apply: lambda { |records, value, _options|
+        if value
```
Comment (CloCkWeRX) [inline app/resources/api/v1/planting_resource.rb:55]: Consider a ternary

**Comment by CloCkWeRX** [comment]
It'd be interesting to maybe land this as two PRs; one adding all of the framework (webpack, etc); the other pulling things out to react components.

It all makes sense; some code climate warnings I'd be keen to see knocked off. Others - long lines, particularly with templates? Maybe we relax a lot of that.

Got screenshots or similar of what the react components give us over the current state?

**Comment by Br3nda** [comment]
I think I'll redo this PR. I was moving the home page to react components to help with our very high memory usage. Might still be worth it, but it's a big change.

**Pull request closed without merging**